src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6933,6 +6935,14 @@ private ExpressionAst HashExpressionRule(Token atCurlyToken, bool parsingSchemaE
             SkipNewlines();
 
             List<KeyValuePair> keyValuePairs = new List<KeyValuePair>();
+
+            bool splatted = false;
+            if (atCurlyToken.Kind == TokenKind.AtAtCurly)
+            {
+                NextToken(); // to skip the first '@' to allow the parsing of the hashtable
```

Comment:
Here where we've seen `AtAtCurly` and we now skip a token, there's a problem because `@@` might be followed by other tokens and we don't check. This causes, for example:

<img width="571" alt="Screen Shot 2019-07-09 at 21 11 36" src="https://user-images.githubusercontent.com/7009879/60939893-49e04100-a28e-11e9-8948-20ed9afa6e35.png">

My recommendation is to restructure the tokens, which I've discussed in a comment in the tokeniser.

But also there are two important things to consider:
- A `@@{` token should only be valid as an argument to a command. This should be a semantic check.
- Adding a new token, error checking and recovery is critical, since otherwise the grammar may have new pitfalls in it. The easiest solution here is to make `@@{` its own whole token.

Reply:
OK, I've changed it to perform a look-ahead so that `@{{` is correctly parsed without ambiguity.